File: examples/core/line-count/src/main/python/LineCount.py
import sys
from pyspark import SparkContext

if __name__ == "__main__":
  if len(sys.argv) < 2 or len(sys.argv) > 3:
    print >> sys.stderr, "Usage: LineCount <path> <search>"
    exit(-1)
  path = sys.argv[1]
  count = 0
  sc = SparkContext()
  if len(sys.argv) == 2:
    count = sc.textFile(path).count()
  else:
    search = sys.argv[2]
    count = sc.textFile(path).filter(lambda line: search in line).count()
  print "Line Count:", count
  sc.stop()

File: examples/core/line-count/src/main/scala/my/spark/core/LineCount.scala
package my.spark.core

import org.apache.spark.SparkContext

object LineCount {
  def main(args: Array[String]) {
    if (args.length < 1 || args.length > 2) {
      System.err.println("Usage: LineCount <path> <filter>")
      System.exit(1)
    }
    var count:Long = 0
    val sc = new SparkContext()
    if (args.length == 1) {
      count = sc.textFile(args(0)).count()
    } else {
      count = sc.textFile(args(0)).filter(_.contains(args(1))).count()
    }
    println( "Line Count: " + count)
    sc.stop()
  }
}
